apps/rush-lib/src/api/BaseFlagFile.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { FileSystem, JsonFile, JsonObject } from '@rushstack/node-core-library';
+
+/**
+ * A helper class for managing last-install flags, which are persistent and
+ * indicate that something installed in the folder was successfully completed.
+ * It also compares state, so that if something like the Node.js version has changed,
+ * it can invalidate the last install.
+ * @internal
+ */
+export abstract class BaseFlagFile {
```

Comment:
Usually the convention is to call base classes `____Base`, so this should be `FlagFileBase`